### PR TITLE
Feature: Widget grouping Phase I (Multi select and Bulk Delete) + Canvas Enhancements.

### LINKED ISSUE #1421: [Enhancement] UI building felt like there was lack of space and grid spaces were too large

### LINKED ISSUE #3798: [Feature] Actions for Widget grouping
Functions/Actions to allow selecting/deselecting multiple widgets.

- [ ] Add actions to select multiple widgets.

Multi-select widgets only if they are siblings.

- [ ] Add checks to multi-select widgets only if they are siblings

### LINKED ISSUE #3801: [Feature] Keyboard Shortcuts for widget grouping
Ability to unselect all selected widgets.

- [ ] Click outside of the selected widgets or on one of them to deselect all the selected widgets.

- [ ] Add Escape shortcut key to deselect all widgets.

- [ ] Add Ctrl + A to select all widgets on canvas

### LINKED ISSUE #4203: [Feature] Allow dismissing property pane with escape
Quite frequently when I'm using Appsmith, the property pane pops up and when I don't want it to. This often happens when I'm rearranging a layout or resizing things. Also, sometimes I click the wrong widget.

It would be nice if I could dismiss the property pane by hitting the escape key, instead of having to click the "x" each time. This would speed things up quite a bit, and make working with Appsmith more fluid. It also intuitively feels like something you should be able to do.

### LINKED ISSUE #4347: [Bug]Property pane overlaps with the left menu section and unable to perform any actions in property pane
## Description
Deploy Preview - https://deploy-preview-4219--appsmith-editor-ce.netlify.app/
It is observed that moving property pane to the left end the property panes hides behind the left menu section and user is not able to perform any actions from property pane.

**Expected Result:**
If the property pane is moved to the left end then property pane should reposition within the canvas area 

**Actual Result:**
Property pane hides behind the left menu section and unable to perform any actions in property pane

### Steps to reproduce the behaviour:
1. Launch the appmsith app in the deploy environment - https://deploy-preview-4219--appsmith-editor-ce.netlify.app/
2. Drag and Drop any widget 
3. Click onEdit 
4. Move the property pane to the left end and observe

**VideoLink:**
https://www.loom.com/share/5fa09f6d2c3c4a5aa826a05281920ae3

### Important Details
 - Version: [Safari - 14.1 (16611.1.21.161.3),Chrome - 90.0.4430.93]
 - OS: [MacOS Big Sur 11.3]
 - Browser [Chrome, Safari]

### LINKED ISSUE #4386: [Bug] Delete button is sometimes inactive in a widget
## Description
1. I've seen this happen a few times within a 10min period, but I'm unable to replicate this bug.
2. See loom video: https://www.loom.com/share/48a6f98b8ee345b98a48a229575a5bad

### LINKED ISSUE #4391: [Bug] Upon Undo action the widget can not be deleted via Entity explorer
Steps to reproduce:
1) Add a widget 
2) Delete the widget using keyboard keys or option on the widget
3) Click on Undo option in the information message 
4) Ensure the widget is placed 
5) Navigate to entity explorer
6) Select the widget 
7) Click on the widget action items (Three dots)
8) Click on Delete and observe 

Observation :
It is observed that the widget does not get deleted from the entity explorer after the UNDO action

Expectation:
The user must be able to delete the widget from entity explorer 

Environment:
Deploy- preview 

Video:
https://www.loom.com/share/ef1d72cd78ab40a8b1066e57b1554a25

### PR DESCRIPTION
**Pull request by marks0351**
## Description
Canvas enhancements - Grid points density, color, adjusting paddings(Might see a slight change in position of widgets coz of these changes)

Widget Grouping
- Select all using CMD + A
- Select multiple widgets CMD + click
- Delete multiple widgets using the DELETE key.
- Undo bulk deletion.
- Hit Esc to clear all selections on the canvas and close the property pane.

Fixes #1421
Fixes #3801
Fixes #3798
Fixes #4203
Fixes #4347
Fixes #4386
Fixes #4391 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce. 
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:green_circle: Total coverage has increased</summary>

    // Code coverage diff between base branch:release and head branch: feature/scaled-grid 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | total | 51.48 **(2.48)** | 32.13 **(3.31)** | 30.1 **(3.25)** | 51.97 **(2.6)**
 :green_circle: | app/client/src/actions/applicationActions.ts | 72 **(4)** | 100 **(0)** | 12.5 **(12.5)** | 58.82 **(5.88)**
 :green_circle: | app/client/src/actions/batchActions.ts | 85.71 **(14.28)** | 100 **(0)** | 50 **(50)** | 100 **(0)**
 :sparkles: :new: | **app/client/src/actions/collectionAction.ts** | **75** | **100** | **0** | **66.67**
 :green_circle: | app/client/src/actions/globalSearchActions.ts | 77.27 **(4.54)** | 100 **(0)** | 28.57 **(14.28)** | 100 **(0)**
 :sparkles: :new: | **app/client/src/actions/importActions.ts** | **75** | **100** | **0** | **66.67**
 :green_circle: | app/client/src/actions/initActions.ts | 90 **(10)** | 100 **(0)** | 66.67 **(33.34)** | 100 **(20)**
 :green_circle: | app/client/src/actions/metaActions.ts | 81.82 **(9.09)** | 100 **(0)** | 33.33 **(33.33)** | 75 **(12.5)**
 :green_circle: | app/client/src/actions/pageActions.tsx | 76.12 **(8.96)** | 0 **(0)** | 33.33 **(28.57)** | 67.44 **(11.63)**
 :sparkles: :new: | **app/client/src/actions/providerActions.ts** | **68** | **100** | **0** | **52.94**
 :sparkles: :new: | **app/client/src/actions/queryPaneActions.ts** | **70** | **100** | **0** | **57.14**
 :sparkles: :new: | **app/client/src/actions/utilActions.ts** | **71.43** | **100** | **0** | **100**
 :green_circle: | app/client/src/actions/widgetActions.tsx | 74.58 **(8.54)** | 0 **(0)** | 27.78 **(27.78)** | 65 **(12.22)**
 :green_circle: | app/client/src/api/ApplicationApi.tsx | 57.5 **(2.5)** | 100 **(0)** | 11.76 **(5.88)** | 61.11 **(2.78)**
 :green_circle: | app/client/src/components/ads/Dropdown.tsx | 62.5 **(41.67)** | 67.11 **(46.06)** | 36.84 **(36.84)** | 59.09 **(36.36)**
 :green_circle: | app/client/src/components/ads/EditableText.tsx | 69.64 **(50.89)** | 41.84 **(31.64)** | 81.82 **(75.76)** | 66.67 **(45.46)**
 :green_circle: | app/client/src/components/ads/Icon.tsx | 58.15 **(1.76)** | 38.95 **(2.11)** | 100 **(0)** | 57.96 **(1.77)**
 :green_circle: | app/client/src/components/ads/Switch.tsx | 100 **(25)** | 100 **(0)** | 100 **(100)** | 100 **(25)**
 :green_circle: | app/client/src/components/ads/Text.tsx | 100 **(4.76)** | 80.77 **(3.85)** | 100 **(0)** | 100 **(4.88)**
 :green_circle: | app/client/src/components/ads/TreeDropdown.tsx | 53.57 **(39.28)** | 58.82 **(47.06)** | 30.43 **(30.43)** | 51.85 **(37.04)**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/CenteredWrapper.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/CopyToClipBoard.tsx** | **30.43** | **33.33** | **0** | **33.33**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/TabbedView.tsx** | **42.86** | **33.33** | **0** | **42.86**
 :sparkles: :new: | **app/client/src/components/designSystems/appsmith/help/CollapsibleHelp.tsx** | **87.5** | **100** | **0** | **87.5**
 :green_circle: | app/client/src/components/designSystems/blueprint/ModalComponent.tsx | 39.13 **(2.77)** | 22.22 **(0)** | 0 **(0)** | 39.13 **(2.77)**
 :green_circle: | app/client/src/components/editorComponents/ActionCreator.tsx | 34.42 **(22.36)** | 23.22 **(23.22)** | 21.11 **(21.11)** | 33.68 **(22.11)**
 :green_circle: | app/client/src/components/editorComponents/DraggableComponent.tsx | 82.14 **(8)** | 58.82 **(17.79)** | 73.33 **(13.33)** | 82 **(8.92)**
 :green_circle: | app/client/src/components/editorComponents/DropTargetComponent.tsx | 62.89 **(1.67)** | 38.36 **(5.45)** | 43.75 **(0)** | 58.14 **(1.32)**
 :green_circle: | app/client/src/components/editorComponents/Dropzone.tsx | 32.56 **(1.61)** | 40 **(0)** | 0 **(0)** | 42.42 **(1.79)**
 :green_circle: | app/client/src/components/editorComponents/EditorContextProvider.tsx | 71.43 **(23.81)** | 100 **(0)** | 40 **(40)** | 71.43 **(23.81)**
 :green_circle: | app/client/src/components/editorComponents/PaneWrapper.tsx | 100 **(60)** | 100 **(0)** | 100 **(100)** | 100 **(60)**
 :green_circle: | app/client/src/components/editorComponents/ResizableComponent.tsx | 56.63 **(1.07)** | 25 **(1.27)** | 60 **(4.44)** | 55.56 **(1.13)**
 :sparkles: :new: | **app/client/src/components/editorComponents/Sidebar.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/DebuggerMessage.tsx** | **77.78** | **100** | **0** | **77.78**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/DebuggerTabs.tsx** | **51.85** | **100** | **0** | **58.33**
 :sparkles: :new: | **app/client/src/components/editorComponents/Debugger/index.tsx** | **90.91** | **50** | **91.67** | **90.91**
 :green_circle: | app/client/src/components/editorComponents/WidgetNameComponent/SettingsControl.tsx | 94.12 **(5.88)** | 85.71 **(14.28)** | 88.89 **(0)** | 94.12 **(5.88)**
 :green_circle: | app/client/src/components/editorComponents/WidgetNameComponent/index.tsx | 81.25 **(3.98)** | 68.42 **(7.13)** | 90 **(1.11)** | 80.43 **(1.86)**
 :sparkles: :new: | **app/client/src/components/editorComponents/form/CredentialsTooltip.tsx** | **90** | **100** | **0** | **90**
 :green_circle: | app/client/src/components/propertyControls/ActionSelectorControl.tsx | 87.5 **(25)** | 66.67 **(66.67)** | 75 **(25)** | 85.71 **(21.42)**
 :green_circle: | app/client/src/components/propertyControls/DropDownControl.tsx | 82.61 **(39.13)** | 44.44 **(44.44)** | 66.67 **(33.34)** | 80 **(35)**
 :green_circle: | app/client/src/components/propertyControls/InputTextControl.tsx | 78.95 **(42.11)** | 31.25 **(31.25)** | 66.67 **(33.34)** | 77.78 **(41.67)**
 :green_circle: | app/client/src/components/propertyControls/StyledControls.tsx | 63.64 **(25.98)** | 76.56 **(14.06)** | 41.67 **(41.67)** | 63.64 **(25.98)**
 :green_circle: | app/client/src/components/propertyControls/SwitchControl.tsx | 92.86 **(21.43)** | 66.67 **(66.67)** | 75 **(25)** | 91.67 **(16.67)**
 :green_circle: | app/client/src/components/utils/TreeStructure.tsx | 100 **(20)** | 100 **(0)** | 100 **(100)** | 100 **(20)**
 :green_circle: | app/client/src/constants/AppConstants.ts | 90.48 **(42.86)** | 50 **(50)** | 100 **(100)** | 94.44 **(50)**
 :sparkles: :new: | **app/client/src/constants/Explorer.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/constants/OmnibarDocumentationConstants.ts** | **30** | **0** | **0** | **30**
 :green_circle: | app/client/src/constants/messages.ts | 70.78 **(0.14)** | 100 **(0)** | 11.54 **(0.43)** | 74.05 **(0.02)**
 :sparkles: :new: | **app/client/src/constants/providerConstants.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/entities/Datasource/RestAPIForm.ts** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/entities/Widget/utils.ts | 96.88 **(3.13)** | 83.87 **(11.29)** | 100 **(18.18)** | 96.77 **(3.22)**
 :green_circle: | app/client/src/icons/ControlIcons.tsx | 55.42 **(2.41)** | 100 **(0)** | 7.5 **(5)** | 55.42 **(2.41)**
 :green_circle: | app/client/src/icons/FormIcons.tsx | 58.82 **(5.88)** | 100 **(0)** | 12.5 **(12.5)** | 58.82 **(5.88)**
 :green_circle: | app/client/src/mockResponses/WidgetConfigResponse.tsx | 28.57 **(0.94)** | 9.38 **(0)** | 28.57 **(0)** | 28.38 **(0.98)**
 :sparkles: :new: | **app/client/src/pages/Editor/EntityNotFoundPane.tsx** | **72.73** | **100** | **0** | **72.73**
 :sparkles: :new: | **app/client/src/pages/Editor/GlobalHotKeys.tsx** | **50.63** | **0** | **40.74** | **49.32**
 :sparkles: :new: | **app/client/src/pages/Editor/MainContainer.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/MainContainerLayoutControl.tsx** | **85.71** | **50** | **50** | **85.71**
 :green_circle: | app/client/src/pages/Editor/Popper.tsx | 85.71 **(21.42)** | 71.74 **(43.48)** | 88.89 **(44.45)** | 85.71 **(21.42)**
 :green_circle: | app/client/src/pages/Editor/PropertyPaneHelpButton.tsx | 85 **(35)** | 50 **(50)** | 50 **(50)** | 84.21 **(31.58)**
 :green_circle: | app/client/src/pages/Editor/PropertyPaneTitle.tsx | 82.69 **(50)** | 38.64 **(20.46)** | 81.25 **(81.25)** | 81.25 **(45.83)**
 :sparkles: :new: | **app/client/src/pages/Editor/WidgetCard.tsx** | **36.11** | **50** | **0** | **40.63**
 :sparkles: :new: | **app/client/src/pages/Editor/WidgetSidebar.tsx** | **34.78** | **21.28** | **0** | **35.82**
 :sparkles: :new: | **app/client/src/pages/Editor/WidgetsEditor.tsx** | **88.89** | **59.38** | **100** | **88.33**
 :sparkles: :new: | **app/client/src/pages/Editor/routes.tsx** | **37.88** | **22.22** | **6.67** | **42.11**
 :green_circle: | app/client/src/pages/Editor/utils.ts | 41.76 **(35.17)** | 12.9 **(12.9)** | 25 **(25)** | 43.53 **(37.65)**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/ApiHomeScreen.tsx** | **37.04** | **24.32** | **3.57** | **37.4**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/CurlImportForm.tsx** | **68.57** | **85.71** | **20** | **74.19**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/ProviderTemplates.tsx** | **35.29** | **37.74** | **4.76** | **37.63**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/RapidApiEditorForm.tsx** | **63.04** | **46.15** | **0** | **63.04**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/helpers.ts** | **75** | **100** | **0** | **66.67**
 :sparkles: :new: | **app/client/src/pages/Editor/APIEditor/index.tsx** | **31.13** | **3.51** | **5** | **33.68**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/BackButton.tsx** | **85.71** | **100** | **0** | **85.71**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/Collapsible.tsx** | **52.63** | **40** | **25** | **60**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/Connected.tsx** | **46.67** | **10.53** | **0** | **47.46**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/DBForm.tsx** | **46.05** | **24** | **7.14** | **45.95**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/DatasourceHome.tsx** | **56.52** | **52.63** | **10** | **58.14**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/DatasourceSection.tsx** | **29.27** | **26.67** | **0** | **28.21**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/FormTitle.tsx** | **27.66** | **5.13** | **0** | **28.89**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/JSONtoForm.tsx** | **15.71** | **20.29** | **4.55** | **16.8**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/RestAPIDatasourceForm.tsx** | **28.49** | **23.4** | **3.23** | **31.25**
 :sparkles: :new: | **app/client/src/pages/Editor/DataSourceEditor/index.tsx** | **28.42** | **0** | **11.76** | **29.07**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/EntityExplorer.tsx** | **83.33** | **47.06** | **40** | **84.44**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/ExplorerSearch.tsx** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/JSDependencies.tsx** | **84.85** | **93.75** | **50** | **89.29**
 :green_circle: | app/client/src/pages/Editor/Explorer/hooks.ts | 45.28 **(30.81)** | 29.23 **(29.23)** | 45.95 **(45.95)** | 48.8 **(36)**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/index.tsx** | **90.91** | **50** | **100** | **90.91**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Onboarding/DBQueryGroup.tsx** | **47.37** | **37.5** | **0** | **48.65**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Onboarding/Loading.tsx** | **71.43** | **100** | **0** | **71.43**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Onboarding/index.tsx** | **46.15** | **50** | **0** | **50**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Pages/PageContextMenu.tsx** | **42.86** | **20** | **0** | **44.12**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Pages/PageEntity.tsx** | **53.33** | **0** | **0** | **53.33**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Pages/PageGroup.tsx** | **64.71** | **5.88** | **33.33** | **63.64**
 :sparkles: :new: | **app/client/src/pages/Editor/Explorer/Widgets/WidgetGroup.tsx** | **65.22** | **6.45** | **0** | **65.22**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/Generator.tsx | 89.47 **(47.36)** | 70 **(70)** | 100 **(100)** | 94.12 **(52.94)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/PropertyControl.tsx | 60.16 **(39.83)** | 40.38 **(40.38)** | 30.77 **(30.77)** | 61.74 **(40)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/PropertyHelpLabel.tsx | 87.5 **(37.5)** | 50 **(50)** | 100 **(100)** | 87.5 **(37.5)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/PropertySection.tsx | 85.71 **(35.71)** | 57.14 **(28.57)** | 66.67 **(66.67)** | 84.62 **(30.77)**
 :green_circle: | app/client/src/pages/Editor/PropertyPane/index.tsx | 81.82 **(32.84)** | 69.89 **(58.06)** | 68 **(48)** | 82.61 **(33.16)**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/AddDatasourceSecurely.tsx** | **91.67** | **100** | **0** | **91.67**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/DatasourceCard.tsx** | **58.97** | **52.94** | **0** | **58.97**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/EditorJSONtoForm.tsx** | **50.4** | **49.48** | **0** | **51.64**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/Form.tsx** | **57.14** | **100** | **0** | **57.14**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/JSONViewer.tsx** | **60** | **60** | **33.33** | **57.89**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/QueryHomeScreen.tsx** | **41.07** | **42.11** | **8.33** | **44.9**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/Table.tsx** | **25** | **10** | **0** | **26.42**
 :sparkles: :new: | **app/client/src/pages/Editor/QueryEditor/index.tsx** | **32.95** | **11.43** | **7.69** | **35**
 :sparkles: :new: | **app/client/src/pages/Editor/SaaSEditor/ActionForm.tsx** | **25.49** | **0** | **0** | **27.08**
 :sparkles: :new: | **app/client/src/pages/Editor/SaaSEditor/DatasourceCard.tsx** | **61.11** | **52.94** | **0** | **61.11**
 :sparkles: :new: | **app/client/src/pages/Editor/SaaSEditor/DatasourceForm.tsx** | **35.42** | **19.05** | **6.25** | **37.5**
 :sparkles: :new: | **app/client/src/pages/Editor/SaaSEditor/ListView.tsx** | **34.18** | **32** | **5.88** | **37.68**
 :green_circle: | app/client/src/pages/Editor/SaaSEditor/constants.ts | 100 **(22.58)** | 60 **(26.67)** | 100 **(25)** | 100 **(25.93)**
 :sparkles: :new: | **app/client/src/pages/Editor/SaaSEditor/routes.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/src/pages/common/NotFound.tsx** | **60** | **33.33** | **0** | **60**
 :green_circle: | app/client/src/reducers/entityReducers/appReducer.ts | 76.92 **(15.38)** | 100 **(0)** | 50 **(33.33)** | 76.92 **(15.38)**
 :green_circle: | app/client/src/reducers/entityReducers/pageListReducer.tsx | 22.22 **(2.78)** | 27.78 **(0)** | 23.08 **(7.7)** | 25 **(3.12)**
 :green_circle: | app/client/src/reducers/evaluationReducers/loadingEntitiesReducer.ts | 85.71 **(14.28)** | 100 **(0)** | 50 **(50)** | 85.71 **(14.28)**
 :green_circle: | app/client/src/reducers/evaluationReducers/treeReducer.ts | 85.71 **(14.28)** | 100 **(0)** | 50 **(50)** | 85.71 **(14.28)**
 :green_circle: | app/client/src/reducers/uiReducers/applicationsReducer.tsx | 10.39 **(2.6)** | 0 **(0)** | 5.71 **(5.71)** | 10.81 **(2.7)**
 :green_circle: | app/client/src/reducers/uiReducers/dragResizeReducer.ts | 57.14 **(11.69)** | 35 **(-65)** | 50 **(50)** | 58.82 **(13.37)**
 :red_circle: | app/client/src/reducers/uiReducers/editorReducer.tsx | 17.91 **(-8.96)** | 0 **(0)** | 18.18 **(4.54)** | 17.91 **(-8.96)**
 :green_circle: | app/client/src/reducers/uiReducers/explorerReducer.ts | 52.63 **(5.26)** | 0 **(0)** | 16.67 **(16.67)** | 52.63 **(5.26)**
 :green_circle: | app/client/src/reducers/uiReducers/propertyPaneReducer.tsx | 62.96 **(44.44)** | 42.86 **(42.86)** | 50 **(50)** | 60 **(40)**
 :green_circle: | app/client/src/sagas/ApplicationSagas.tsx | 20.12 **(1.78)** | 3.92 **(1.96)** | 10.53 **(5.27)** | 23.61 **(2.08)**
 :green_circle: | app/client/src/sagas/BatchSagas.tsx | 41.94 **(16.13)** | 25 **(16.67)** | 40 **(20)** | 52 **(20)**
 :green_circle: | app/client/src/sagas/EvaluationsSaga.ts | 20.83 **(0)** | 1.92 **(0)** | 10 **(5)** | 23.62 **(0)**
 :green_circle: | app/client/src/sagas/GlobalSearchSagas.ts | 51.28 **(28.2)** | 13.04 **(8.69)** | 50 **(33.33)** | 52.94 **(26.47)**
 :green_circle: | app/client/src/sagas/InitSagas.ts | 39.62 **(15.09)** | 3.33 **(0)** | 60 **(40)** | 43.96 **(15.39)**
 :green_circle: | app/client/src/sagas/OnboardingSagas.ts | 18.08 **(1.87)** | 3.57 **(2.14)** | 12.5 **(4.17)** | 18.39 **(2.21)**
 :green_circle: | app/client/src/sagas/PageSagas.tsx | 17.63 **(0.11)** | 3.39 **(-0.12)** | 9.68 **(3.23)** | 20.17 **(0.08)**
 :green_circle: | app/client/src/sagas/RecentEntitiesSagas.ts | 44.83 **(17.24)** | 10.61 **(9.09)** | 50 **(25)** | 44.83 **(17.24)**
 :green_circle: | app/client/src/sagas/WidgetEnhancementHelpers.ts | 51.39 **(23.61)** | 36.17 **(31.91)** | 42.86 **(35.72)** | 54.69 **(25)**
 :green_circle: | app/client/src/sagas/WidgetLoadingSaga.ts | 26.32 **(7.02)** | 21.43 **(14.29)** | 15.38 **(7.69)** | 30 **(8)**
 :green_circle: | app/client/src/sagas/WidgetOperationSagas.tsx | 11.95 **(3.85)** | 3.46 **(2.52)** | 9.33 **(6.25)** | 12.68 **(3.44)**
 :green_circle: | app/client/src/sagas/index.tsx | 90.91 **(0.21)** | 60 **(10)** | 100 **(0)** | 92.86 **(0)**
 :green_circle: | app/client/src/sagas/selectors.tsx | 61.8 **(16.35)** | 36.36 **(36.36)** | 28 **(28)** | 55.22 **(19.51)**
 :green_circle: | app/client/src/selectors/applicationSelectors.tsx | 65.75 **(1.37)** | 27.27 **(0)** | 40.91 **(4.55)** | 57.89 **(1.75)**
 :green_circle: | app/client/src/selectors/editorSelectors.tsx | 61.54 **(2.31)** | 43.24 **(16.21)** | 42.86 **(7.15)** | 57.94 **(0.93)**
 :green_circle: | app/client/src/selectors/entitiesSelector.ts | 44.75 **(0.46)** | 2.86 **(1.43)** | 14.81 **(0)** | 41.61 **(0.62)**
 :green_circle: | app/client/src/selectors/propertyPaneSelectors.tsx | 80 **(17.04)** | 76.19 **(30.04)** | 100 **(0)** | 79.31 **(17.77)**
 :sparkles: :new: | **app/client/src/selectors/ui.ts** | **100** | **100** | **100** | **100**
 :green_circle: | app/client/src/selectors/widgetSelectors.ts | 88.24 **(41.18)** | 50 **(50)** | 83.33 **(83.33)** | 92.86 **(35.72)**
 :sparkles: :new: | **app/client/src/transformers/RestAPIDatasourceFormTransformer.ts** | **18.64** | **0** | **0** | **17.65**
 :green_circle: | app/client/src/utils/AppsmithUtils.tsx | 54.19 **(4.19)** | 21.95 **(6.1)** | 40.54 **(4.43)** | 49.68 **(4.58)**
 :green_circle: | app/client/src/utils/DynamicBindingUtils.ts | 88.46 **(1.92)** | 75.41 **(4.92)** | 78.57 **(7.14)** | 87.1 **(2.15)**
 :green_circle: | app/client/src/utils/PerformanceTracker.ts | 53.21 **(11.93)** | 11.69 **(7.79)** | 60 **(30)** | 52.78 **(11.11)**
 :green_circle: | app/client/src/utils/PropertyControlFactory.tsx | 70 **(30)** | 33.33 **(33.33)** | 75 **(25)** | 73.68 **(31.57)**
 :green_circle: | app/client/src/utils/PropertyControlRegistry.tsx | 100 **(9.09)** | 100 **(0)** | 100 **(25)** | 100 **(9.09)**
 :green_circle: | app/client/src/utils/WidgetPropsUtils.tsx | 64.96 **(4.55)** | 50.14 **(1.18)** | 53.33 **(3.33)** | 64.27 **(4.65)**
 :green_circle: | app/client/src/utils/hooks/dragResizeHooks.tsx | 83.67 **(31.59)** | 37.5 **(37.5)** | 70.59 **(45.59)** | 80.49 **(32.99)**
 :green_circle: | app/client/src/utils/hooks/useClickOpenPropPane.tsx | 96.43 **(8.43)** | 86.67 **(46.67)** | 100 **(0)** | 96.3 **(9.34)**
 :sparkles: :new: | **app/client/src/utils/hooks/useDynamicAppLayout.tsx** | **94.23** | **52.17** | **100** | **93.48**
 :green_circle: | app/client/src/widgets/BaseWidget.tsx | 80.51 **(0)** | 41.86 **(2.84)** | 83.33 **(3.33)** | 79.46 **(0)**
 :green_circle: | app/client/src/widgets/InputWidget.tsx | 70.18 **(0.54)** | 34.38 **(0)** | 58.33 **(0)** | 76.47 **(0.47)**
 :green_circle: | app/client/src/widgets/MetaHOC.tsx | 82.81 **(6.25)** | 58.82 **(5.88)** | 92.31 **(0)** | 83.64 **(7.28)**
 :green_circle: | app/client/src/widgets/Tabs/TabsWidget.tsx | 66.42 **(0.25)** | 36.56 **(0)** | 60.71 **(0)** | 65.35 **(0.27)**
 :sparkles: :new: | **app/client/test/sagas.ts** | **100** | **100** | **100** | **100**
 :sparkles: :new: | **app/client/test/testCommon.tsx** | **100** | **50** | **100** | **100**
 :red_circle: | app/client/test/testUtils.tsx | 95 **(-5)** | 85.71 **(-5.96)** | 100 **(0)** | 97.3 **(-2.7)**
 :green_circle: | app/client/test/factories/WidgetFactoryUtils.ts | 94.44 **(1.58)** | 50 **(0)** | 100 **(0)** | 93.75 **(1.44)**
 :green_circle: | app/client/test/factories/Widgets/ContainerFactory.ts | 100 **(25)** | 100 **(0)** | 100 **(100)** | 100 **(25)**
 :green_circle: | app/client/test/factories/Widgets/FormFactory.ts | 100 **(25)** | 100 **(0)** | 100 **(100)** | 100 **(25)**
 :x: | ~~app/client/src/selectors/ui.tsx~~ | ~~100~~ | ~~100~~ | ~~100~~ | ~~100~~</details>

### CODE REVIEW

**Review by jsartisan — DISMISSED** [review]

File: app/client/src/sagas/selectors.tsx [hunk 1]
```diff
@@ -123,7 +123,27 @@ export const getPluginIdOfPackageName = (
...
 
 export const getSelectedWidget = (state: AppState) => {
-  const selectedWidgetId = state.ui.widgetDragResize.selectedWidget;
+  const selectedWidgetId = state.ui.widgetDragResize.lastSelectedWidget;
   if (!selectedWidgetId) return;
   return state.entities.canvasWidgets[selectedWidgetId];
 };
+
+export const getWidgetChildren = createSelector(
+  getWidget,
```
Comment (riodeuno) [inline app/client/src/sagas/selectors.tsx:132]: Let's add a toast when we select all `All widgets in this page including modals have been selected`

File: app/client/src/selectors/ui.tsx [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import { AppState } from "reducers";
 
 export const getSelectedWidget = (state: AppState) =>
```
Comment (riodeuno) [inline app/client/src/selectors/ui.tsx:3]: Let's rename this to .ts

File: app/client/src/utils/hooks/useClickOpenPropPane.tsx [hunk 3]
```diff
@@ -26,15 +29,25 @@ export const useClickOpenPropPane = () => {
...
+      appMode !== APP_MODE.EDIT ||
+      targetWidgetId !== focusedWidget
+    )
+      return;
     if (
       (!isPropPaneVisible && selectedWidgetId === focusedWidget) ||
       selectedWidgetId !== focusedWidget
     ) {
-      selectWidget(focusedWidget);
+      const isMultiSelect = e.metaKey || e.ctrlKey;
```
Comment (riodeuno) [inline app/client/src/utils/hooks/useClickOpenPropPane.tsx:45]: Let's see if we can make this very specific when identifying.

File: app/client/src/widgets/ModalWidget.tsx [hunk 4]
```diff
@@ -20,11 +20,11 @@ import { getWidget } from "sagas/selectors";
 const MODAL_SIZE: { [id: string]: { width: number; height: number } } = {
   MODAL_SMALL: {
     width: 456,
-    height: GridDefaults.DEFAULT_GRID_ROW_HEIGHT * 6,
+    height: GridDefaults.DEFAULT_GRID_ROW_HEIGHT * 24,
```
Comment (riodeuno) [inline app/client/src/widgets/ModalWidget.tsx:23]: Let's verify this.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by riodeuno — DISMISSED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by jsartisan — APPROVED** [review]

**Review by SatishGandham — APPROVED** [review]
Approving to run tests.

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — DISMISSED** [review]

**Review by riodeuno — DISMISSED** [review]

**Review by riodeuno — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Pull request merged**